src/hooks.ts

```diff
@@ -33,6 +34,30 @@ export const createAPIHooks = <Endpoints extends RoughEndpoints>({
         options,
       );
     },
+    useAPIInfiniteQuery: (route, initPayload, options) => {
+      const { nextPageParamKey } = initPayload;
+      // @ts-expect-error, typescript enforcing only deleting non-required props
+      delete initPayload.nextPageParamKey;
```

Comment:
To avoid a mutation (and the type error) here, we can do this:

`const { nextPageParamKey, ...cleanPayload } = initPayload;`

and then just use `cleanPayload` instead of `initPayload`.

This also keeps the function pure. I think mutating the argument could definitely cause unwanted side effects.